packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1347,22 +1368,28 @@ async function renderToStream(
     onHTMLRenderSSRError
   )
 
-  let primaryRenderReactServerStream: null | ReadableStream<Uint8Array> = null
+  let reactServerResult: null | ReactServerResult = null
 
   const setHeader = res.setHeader.bind(res)
 
   try {
     // This is a dynamic render. We don't do dynamic tracking because we're not prerendering
     const RSCPayload = await getRSCPayload(tree, ctx, res.statusCode === 404)
-    const reactServerStream = ComponentMod.renderToReadableStream(
-      RSCPayload,
-      clientReferenceManifest.clientModules,
-      {
-        onError: serverComponentsErrorHandler,
-        nonce: ctx.nonce,
-      }
+    reactServerResult = new ReactServerResult(
+      ComponentMod.renderToReadableStream(
+        RSCPayload,
+        clientReferenceManifest.clientModules,
+        {
+          onError: serverComponentsErrorHandler,
+          nonce: ctx.nonce,
+        }
+      )
     )
-    primaryRenderReactServerStream = reactServerStream
+
+    // React doesn't start rendering synchronously but we want the RSC render to have a chance to start
+    // before we begin SSR rendering because we want to capture any available preload headers so we tick
+    // one task before continuign
```

Comment:
sorry I'm grammar police I just tend to read all of these comments 😆 

```suggestion
    // one task before continuing
```